src/blocking/i2c.rs

```diff
@@ -125,3 +125,92 @@ pub trait WriteIterRead {
     where
         B: IntoIterator<Item = u8>;
 }
+
+/// Transactional I2C operation.
+///
+/// Several operations can be combined as part of a transaction.
+#[derive(Debug, PartialEq)]
+pub enum Operation<'a> {
+    /// Read data into the provided buffer
+    Read(&'a mut [u8]),
+    /// Write data from the provided buffer
+    Write(&'a [u8]),
+}
+
+/// Transactional I2C interface.
+///
+/// This allows combining operations within an I2C transaction.
+pub trait Transactional {
+    /// Error type
+    type Error;
+
+    /// Execute the provided operations on the I2C bus.
+    ///
+    /// Transaction contract:
+    /// - Before executing the first operation an ST is sent automatically. This is followed by SAD+R/W as appropriate.
```

Comment:
is any customisation of this contract required, or is this order and combinations of operations always valid?

Reply:
From my understanding, any order and combination should be valid. See I2C specification sections [3.2.4](https://www.nxp.com/docs/en/user-guide/UM10204.pdf#G1672579), [3.2.7](https://www.nxp.com/docs/en/user-guide/UM10204.pdf#G1672934) and also [Wikipedia](https://en.wikipedia.org/wiki/I%C2%B2C#Transaction_format).